source/SharedCrtResourceManager.cpp

```diff
@@ -252,7 +235,10 @@ int SharedCrtResourceManager::establishConnection(const PlainConfig &config)
     }
     auto clientConfigBuilder = MqttClientConnectionConfigBuilder(config.cert->c_str(), config.key->c_str());
     clientConfigBuilder.WithEndpoint(config.endpoint->c_str());
-    clientConfigBuilder.WithCertificateAuthority(config.rootCa->c_str());
+    if (config.rootCa.has_value())
```

Comment:
Also add `not empty` check.

Reply:
I can add but rootCa will not have a value if it is passed as an empty string. Lots of tests to ensure that behavior.

RootCa is also optional now and this would fix the segmentation fault bug. The default behavior of the SDK is to use default trust store if no rootCa. What core functionality does this change?